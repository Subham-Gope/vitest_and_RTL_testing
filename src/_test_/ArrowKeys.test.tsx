// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ArrowKeys from "../components/ArrowKeys";

// describe("Testing the ArrowKey.tsx file", () => {
//   test("displays the pressed arrow key", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(<ArrowKeys message="testing" />);
//     expect(container).toMatchSnapshot();

//     const input = screen.getByLabelText(/enter/i);
//     expect(input).toBeInTheDocument();

//     // initiating user interaction
//     input.focus();
//     await user.keyboard("{ArrowDown}");

//     const displayText = screen.getByText(/pressed:/i);
//     expect(displayText).toBeInTheDocument();

//     expect(displayText).toHaveTextContent("Pressed: ArrowDown");
//   });
// });
