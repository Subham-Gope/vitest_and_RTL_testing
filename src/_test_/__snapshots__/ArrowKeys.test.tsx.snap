// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ArrowKey.tsx file > displays the pressed arrow key 1`] = `
<div>
  <h1>
    testing
  </h1>
  <div>
    <label
      for="arrow"
    >
      Enter the text
    </label>
    <input
      id="arrow"
      placeholder="Enter your text"
      type="text"
    />
  </div>
</div>
`;
